### PR TITLE
Update spring cloud sample for bug modification and api upgrade

### PR DESCRIPTION
**Pull request by moarychan**
Issue: [Error found when running Spring samples of Event Hubs and should use Sinks API instead of EmitterProcessor](https://github.com/Azure/azure-sdk-for-java/issues/20111)

Changelogs
- Use `Sinks` api instead of the deprecated `EmitterProcessor`， involving the EventHubs and Service Bus samples.
- Use logger instead `System.out.print`.
- Use `com.azure:azure-core:1.14.1` package to fix conflicts for Event Hubs sample.

The to-do item "Update event hubs and service bus sample test case" will be processed in a new PR.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java [hunk 1]
```diff
@@ -49,9 +52,9 @@ private void sendQueueMessage() throws InterruptedException {
         final String messageBody = "queue message";
 
         queueSender.sendMessage(new ServiceBusMessage(BinaryData.fromBytes(messageBody.getBytes(UTF_8)))).subscribe(
-            v -> System.out.println("Sent message: " + messageBody),
-            e -> System.err.println("Error occurred while sending message: " + e),
-            () -> System.out.println("Send message to queue complete.")
+            v -> LOGGER.info("Sent message: " + messageBody),
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java:55]: if we are using LOGGER, we should use it like `Sent message: {}`

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java [hunk 2]
```diff
@@ -49,9 +52,9 @@ private void sendQueueMessage() throws InterruptedException {
...
+            e -> LOGGER.error("Error occurred while sending message", e),
+            () -> LOGGER.info("Send message to queue complete.")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java:57]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java [hunk 3]
```diff
@@ -61,7 +64,7 @@ private void sendQueueMessage() throws InterruptedException {
 
     private void receiveQueueMessage() throws InterruptedException {
         queueReceiver.receiveMessages().subscribe(message ->
-            System.out.println("Received Message: " + message.getBody().toString()));
+            LOGGER.info("Received Message: " + message.getBody().toString()));
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java:67]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java [hunk 4]
```diff
@@ -72,9 +75,9 @@ private void sendTopicMessage() throws InterruptedException {
         final String messageBody = "topic message";
 
         topicSender.sendMessage(new ServiceBusMessage(BinaryData.fromBytes(messageBody.getBytes(UTF_8)))).subscribe(
-            v -> System.out.println("Sent message: " + messageBody),
-            e -> System.err.println("Error occurred while sending message: " + e),
-            () -> System.out.println("Send message to topic complete.")
+            v -> LOGGER.info("Sent message: " + messageBody),
+            e -> LOGGER.error("Error occurred while sending message", e),
+            () -> LOGGER.info("Send message to topic complete.")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java:80]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java [hunk 5]
```diff
@@ -84,7 +87,7 @@ private void sendTopicMessage() throws InterruptedException {
 
     private void receiveSubscriptionMessage() throws InterruptedException {
         topicSubscriber.receiveMessages().subscribe(message ->
-            System.out.println("Received Message: " + message.getBody().toString()));
+            LOGGER.info("Received Message: " + message.getBody().toString()));
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-boot-sample-servicebus/src/main/java/com/azure/spring/sample/servicebus/ServiceBusSampleApplication.java:90]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml [hunk 6]
```diff
@@ -12,7 +12,12 @@
   <artifactId>azure-spring-cloud-sample-eventhubs-binder</artifactId>  
   <groupId>com.azure.spring</groupId>  
   <version>1.0.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-sample-eventhubs-binder;current} -->
-  <name>Azure Spring Cloud Sample Event Hubs Binder</name>  
+  <name>Azure Spring Cloud Sample Event Hubs Binder</name>
+
+  <properties>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:17]: why adding this?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml [hunk 7]
```diff
@@ -74,7 +84,19 @@
...
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>5.7.1</version> <!-- {x-version-update;org.junit.jupiter:junit-jupiter-engine;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
+      <version>5.7.1</version> <!-- {x-version-update;org.junit.vintage:junit-vintage-engine;external_dependency} -->
+      <scope>test</scope>
+    </dependency>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:99]: Why are these two needed?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml [hunk 8]
```diff
@@ -108,7 +130,23 @@
             </resources>
           </deployment>
         </configuration> 
-      </plugin> 
+      </plugin>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:133]: why adding this

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml [hunk 9]
```diff
@@ -17,6 +17,8 @@ spring:
         checkpoint-storage-account: [checkpoint-storage-account]
         checkpoint-container: [checkpoint-container]
     stream:
+      function:
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml:20]: why move this here?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-sp.yaml [hunk 10]
```diff
@@ -18,6 +18,8 @@ spring:
         checkpoint-storage-account: [checkpoint-storage-account]
         checkpoint-container: [checkpoint-container]
     stream:
+      function:
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-sp.yaml:21]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml [hunk 11]
```diff
@@ -8,6 +8,8 @@ spring:
         checkpoint-container: [checkpoint-container]
 
     stream:
+      function:
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application.yaml:11]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplicationIT.java [hunk 12]
```diff
@@ -24,7 +24,7 @@
 @RunWith(SpringRunner.class)
 @SpringBootTest(classes = EventHubBinderApplication.class)
 @AutoConfigureMockMvc
-@TestPropertySource(locations = "classpath:application-test.properties")
+@ActiveProfiles("manual")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplicationIT.java:27]: I thought we are going to make another PR for the integration tests?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/pom.xml [hunk 13]
```diff
@@ -14,6 +14,10 @@
   <version>1.0.0</version> <!-- {x-version-update;com.azure.spring:azure-spring-cloud-sample-eventhubs-kafka;current} -->
   <name>Azure Spring Cloud Sample Event Hubs Kafka</name>
 
+  <properties>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/pom.xml:17]: why are we changing this

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/pom.xml [hunk 14]
```diff
@@ -71,5 +80,37 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <version>2.22.0</version> <!-- {x-version-update;org.apache.maven.plugins:maven-failsafe-plugin;external_dependency} -->
+        <executions>
+          <execution>
+            <id>default</id>
+            <goals>
+              <goal>integration-test</goal>
+            </goals>
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/pom.xml:107]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/main/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplication.java [hunk 15]
```diff
@@ -3,16 +3,44 @@
...
+    @Bean
+    public Sinks.Many<Message<String>> many() {
+        return Sinks.many().unicast().onBackpressureBuffer();
+    }
+
+    @Bean
+    public Supplier<Flux<Message<String>>> supply(Sinks.Many<Message<String>> many) {
+        return () -> many.asFlux()
+                         .doOnNext(m -> LOGGER.info("Manually sending message {}", m))
+                         .doOnError(t -> LOGGER.error("Error encountered", t));
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/main/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplication.java:39]: same here

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/test/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplicationIT.java [hunk 16]
```diff
@@ -24,7 +24,7 @@
 @RunWith(SpringRunner.class)
 @SpringBootTest(classes = EventHubKafkaBinderApplication.class)
 @AutoConfigureMockMvc
-@TestPropertySource(locations = "classpath:application-test.properties")
+@ActiveProfiles("test")
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/test/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplicationIT.java:27]: why this

**Review by saragluna — CHANGES_REQUESTED** [review]
How about we create another PR for the integration test changes?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml (hunk 6 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:17]: Otherwise, test IT will not be found to execute, also need 'org.junit.jupiter' dependencies.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml (hunk 7 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/pom.xml:99]: It will cause the test cases not to be found.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml (hunk 9 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml:20]: Hope this is clearer, I put it at the bottom level of spring.stream.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/test/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplicationIT.java (hunk 16 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-kafka/src/test/java/com/azure/spring/sample/eventhubs/kafka/EventHubKafkaBinderApplicationIT.java]: It can be removed.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplicationIT.java (hunk 12 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/test/java/com/azure/spring/sample/eventhubs/binder/EventHubBinderApplicationIT.java]: Reverted.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventProducerController.java [hunk 17]
```diff
@@ -26,13 +26,13 @@
...
-    private EmitterProcessor<Message<String>> emitterProcessor;
+    private Sinks.Many<Message<String>> many;
 
     @PostMapping("/messages")
     public ResponseEntity<String> sendMessage(@RequestParam String message) {
-        LOGGER.info("Going to add message {} to emitter", message);
-        emitterProcessor.onNext(MessageBuilder.withPayload(message).build());
-        return ResponseEntity.ok("Sent!");
+        LOGGER.info("Going to add message {} to Sinks.Many.", message);
+        many.emitNext(MessageBuilder.withPayload(message).build(), Sinks.EmitFailureHandler.FAIL_FAST);
```
Comment (saragluna) [inline sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventProducerController.java:34]: could you explain more on this method?

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml (hunk 9 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/resources/application-mi.yaml:20]: okay, even though it makes no difference to me.

**Review by saragluna — APPROVED** [review]
LGTM.

File: sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventProducerController.java (hunk 17 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-samples/azure-spring-cloud-sample-eventhubs-binder/src/main/java/com/azure/spring/sample/eventhubs/binder/EventProducerController.java:34]: You can some code snippet in this post https://spring.io/blog/2020/07/10/first-milestone-of-reactor-2020-0-codename-europium
